anthunder/client/aio_client.py

```diff
@@ -143,9 +144,22 @@ def _inner(fut):
 
     async def _get_connection(self, address):
         try:
-            reader, writer = await asyncio.open_connection(*address)
-            task = asyncio.ensure_future(self._recv_response(reader, writer))
-            return self.connection_mapping.setdefault(address, (task, writer))
+            # fast path return existed connection
+            ret = self.connection_mapping.get(address)
+            if ret:
+                return ret
+
+            lock = self._pending_dial.get(address)
```

Comment:
A `setdefault` is enough here for the next few lines.

Reply:
got